private/bufpkg/bufcheck/internal/bufcheckserver/internal/bufcheckserverhandle/lint.go

```diff
@@ -960,17 +960,15 @@ func handleLintMessageProtovalidate(
 			args...,
 		)
 	}
-	// TODO: make buflintvalidate.CheckMessage's signature the same as this function,
-	// instead of accepting a legacy addFunc as an argument. Same comment for CheckField.
```

Comment:
Talked to @oliversun09 who added this TODO — the goal here is not to refactor the check logic, so porting the current use of ResponseWriter to the original addFunc signature is fine, especially since `buflintvalidate` retains that signature internally. Added some comments and a small rename to addAnnotationFunc for clarity at the call sites (CheckMessage and CheckField).